### PR TITLE
In release/1.1.0, use 1.1.0 buildinfo dependency

### PR DESCRIPTION
**Pull request by dagood**
This changes dependencies.props in release/1.1.0 to depend on 1.1.0 builds of corefx and coreclr.

I had to run `UpdateDependencies` and I separated that into its own commit.

This allows release/1.1.0 subscriptions from coreclr and corefx. (The coreclr subscription is for testing against a coreclr package.)

(Similar corefx PR: https://github.com/dotnet/corefx/pull/11533)

@gkhanna79 @wtgodbe

### CODE REVIEW

**Comment by dagood** [comment]
I made some changes in how the runtime versions are propagated: full versions in dependencies.props. Getting them from dependencies.props puts them in one place for easy changes, and can be used for auto-update (or semi-auto update, where someone changes the buildinfos and runs `UpdateDependencies`.)

The other option is to put the stable.minor.patch in the pkgproj's themselves and just delete the `-$(ExternalExpectedPrerelease)` part.

/cc @weshaggard

File: dependencies.props [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
 
   <!-- Full package version strings that are used in other parts of the build. -->
   <PropertyGroup>
-    <CoreClrPackageVersion>1.1.0-beta-24431-01</CoreClrPackageVersion>
+    <CoreClrPackageVersion>1.1.0-beta-24506-03</CoreClrPackageVersion>
     <XunitPackageVersion>2.1.0</XunitPackageVersion>
+
+    <!-- Runtime package versions from projectk-tfs/release/1.0.0 -->
+    <RuntimeWin7ArmMicrosoftNETCoreTestHostPackageVersion>1.0.0<RuntimeWin7ArmMicrosoftNETCoreTestHostPackageVersion>
+    <RuntimeWin8ArmMicrosoftNETCoreRuntimeCoreCLRPackageVersion>1.0.0</RuntimeWin8ArmMicrosoftNETCoreRuntimeCoreCLRPackageVersion>
```
Comment (gkhanna79) [inline dependencies.props]: I think we should reference the latest package versions which are 1.0.4.

Comment (dagood) [inline dependencies.props]: Interesting, doesn't seem to be listed in dotnet/versions. I'll update this.

**Comment by dagood** [comment]
@dotnet-bot test this please

**Comment by dagood** [comment]
An update on the errors here, although I'm still working through solving them:

Through the `build` command (rather than `build-tests.cmd`), `CORE_ROOT` doesn't seem to be populated with a `.nuget/pkg` folder. When `coreclr\tests\src\Common\test_dependencies\project.json` is built, it runs [`RestorePackages`](https://github.com/dotnet/buildtools/blob/3e69337852e5b1607c98591c7cc45188e4e8801f/src/Microsoft.DotNet.Build.Tasks/PackageFiles/packageresolve.targets#L25-L36). That task first checks to see if the project.json needs restore (has an old lockfile or some packages are missing from the packages dir). If it needs restore, then it restores them. The error here is that we add the [`$(CORE_ROOT)\.nuget\pkg`](https://github.com/dotnet/coreclr/blob/610952b412bbaa2eb568d32211253153e040cddf/tests/src/Common/test_dependencies/test_dependencies.csproj#L10) folder to the source list, which doesn't exist because we used `build`.

It seems like this error should happen whenever test_dependencies needs to be restored. I think the src-wide restore hides the problem because it convinces the restore task that no restore is necessary.

I'm not sure yet why this PR is making the restore command think that a test_dependencies restore is needed. Looking into that.

@ramarag, when you were making the change to unify these dependencies, do you know the logic behind adding `$(CORE_ROOT)\.nuget\pkg` to the source list? It would be nice to just remove it, or make sure it's only added when necessary.

**Comment by dagood** [comment]
I added a change to only try to use `$(CORE_ROOT)\.nuget\pkg` if it exists. It's odd that there was this hard dependency on a folder that seems like it usually wouldn't be there. Let me know if I misunderstood what's going on there.

**Comment by dagood** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build ([segfault](http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/release_1.1.0/job/arm_emulator_cross_debug_ubuntu_prtest/6/))

**Comment by dagood** [comment]
Hmm. [corerun error `0x80004005`](http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/release_1.1.0/job/arm_emulator_cross_debug_ubuntu_prtest/7/)on linux ARM build--I remember seeing that with mismatched dll versions earlier. I don't know how to repro the Linux ARM build and potentially see what the mismatches are, though.
```
FAILED   - JIT/Regression/CLR-x86-JIT/V1-M12-Beta2/b80045/b80045/b80045.sh
    BEGIN EXECUTION
    /bindings/tmp/arm32_ci_temp/coreclr/Windows_NT.x64.Debug/Tests/coreoverlay/corerun b80045.exe
    coreclr_initialize failed - status: 0x80004005
    Expected: 100
    Actual: 255
    END EXECUTION - FAILED
```
Just in case, @dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by dagood** [comment]
Third time's the charm, apparently. CI is green.

@wtgodbe @ramarag Mind looking this over?

**Comment by wtgodbe** [comment]
LGTM, but I'd wait on Rama

**Comment by gkhanna79** [comment]
@ramarag PTAL at this at earliest since this is required to unblock 1.1 package publishing.

**Comment by ramarag** [comment]
@dagood: The property you are talking about was added as part of https://github.com/dotnet/coreclr/pull/3387 meant for helix integration

@wtgodbe : I suspect you are not using it any more, if that is the case you can remove it

**Comment by ramarag** [comment]
otherwise it LGTM

**Comment by dagood** [comment]
Thanks! Leaving the property in with that condition for now. Maybe we could have an issue to track removing it entirely (if that is possible).

**Pull request merged**